codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/nameresolver/AwsSdkNativeV2.java

```diff
@@ -162,7 +152,7 @@ private TypeName typeForShapeNoEnum(ShapeId shapeId) {
     final Shape shape = model.expectShape(shapeId);
 
     if (shape.hasTrait(EnumTrait.class)) {
-      if (shapeRequiresTypeConversionFromStringToStructure(shapeId)) {
+      if (true || shapeRequiresTypeConversionFromStringToStructure(shapeId)) {
```

Comment:
An `or` with a constant true is illogical. 

Is there a TODO here? Or should we just remove the if?

Reply:
Implicit TODO, I was experimenting with whether we needed this test anymore. Will resolve before marking the PR actually ready :)

Reply:
Yeah... there was an edge case with Enums and the Java SDK
that I used this to work around.

I swear I described it somewhere in Java codegen...